src/debug/todoapp-test.js

```diff
@@ -0,0 +1,176 @@
+                lines: [69]
+            }
+
+        ];
+    }
+
+    withEngine(engine) {
+        const outputList = [];
+
+        let postData = {
+            "description": "Breakfast" + randomStr,
+            "owner": "barney",
+            "finish": "false"
+        };
+        let postRequest = {
+            url: url + '/api/todolist',
+            port: 8080,
+            method: 'POST',
+            body: postData,
+            json: true
+        };
+        engine.registerHandler('breakpoint:*/TodoApplication.java:*', async (event, arg1, arg2, detail) => {
+            const breakpointFile = path.join(engine.cwd, this.sourcePath, 'com/microsoft/azure/sample/TodoApplication.java');
+            const expectedLines = [15];
+            utils.pathEquals(breakpointFile, detail.source.path).should.equal(true);
+            console.log("### Hit breakPoint:15");
+            console.log('***threads', await engine.threads());
+            const scopes = await engine.scopes(detail.id);
+            await engine.resume(detail.event.body.threadId);
+
+        });
+
+        engine.registerHandler('breakpoint:*/TodoListController.java:*', async (event, arg1, arg2, detail) => {
+            const breakpointFile = path.join(engine.cwd, this.sourcePath, 'com/microsoft/azure/sample/controller/TodoListController.java');
+            const expectedLines = [69];
+            console.log("### Hit #breakPoint:69");
+            utils.pathEquals(breakpointFile, detail.source.path).should.equal(true);
+            await engine.resume(detail.event.body.threadId);
+
+            //wait one second to let the app handle the post request
+            await utils.timeout(1000);
+            console.log("Send Get Request");
+            let getRes = syncRequest('GET', url + '/api/todolist');
+            let getResBody = JSON.parse(getRes.getBody('utf8'));
+            console.log(getResBody);
+            let descriptions = [];
+            for (let index in getResBody) {
+                descriptions.push(getResBody[index].description);
+            }
+            console.log(descriptions);
+            let index = descriptions.indexOf("Breakfast" + randomStr);
+            (index >= 0).should.equal(true);
+            //delete the item
+            let deleteResult = syncRequest('DELETE', url + '/api/todolist/' + getResBody[index].id).getBody('utf8');
+            console.log("Delete result:" + deleteResult);
+            await engine.disconnect(false);
+        });
+
+        engine.registerHandler('output*', (event, arg1, arg2, detail) => {
+            detail.category.should.equal('stdout');
+            outputList.push(detail.output);
+            console.log("****", detail.output);
+            if (detail.output.includes("Started TodoApplication")) {
+                console.log("send request to hit second BP");
+                request(postRequest, function (err, res, body) {
```

Comment:
1. functions to =>
2. you need to wrap the request(postRequest,...) into a promise, otherwise the throw err will not cause the test failure.

Reply:
Got it